measure-builds/src/main/java/com/automattic/android/measure/reporters/InternalWooCommerceA8cCiReporter.kt

```diff
@@ -22,38 +20,38 @@ import io.ktor.http.HttpHeaders
 import io.ktor.http.HttpHeaders.Authorization
 import io.ktor.http.HttpStatusCode
 import io.ktor.http.contentType
-import kotlinx.serialization.encodeToString
-import kotlinx.serialization.json.Json
-import org.gradle.api.logging.Logger
-import org.gradle.api.provider.Provider
+import kotlinx.coroutines.runBlocking
 import java.io.File
 import java.util.Locale
 import java.util.concurrent.TimeUnit.MILLISECONDS
 import java.util.concurrent.TimeUnit.MINUTES
-import kotlin.io.path.Path
-import kotlin.io.path.absolutePathString
-import kotlin.io.path.createDirectories
-import kotlin.io.path.createFile
-import kotlin.io.path.exists
-import kotlin.io.path.writeText
 import kotlin.time.Duration.Companion.seconds
 
-class MetricsReporter(
-    private val logger: Logger,
-    private val authToken: Provider<String>,
-    private val buildDir: File,
-) {
-    suspend fun report(
-        report: InMemoryReport,
-        gradleScanId: String?
+
+object InternalA8cCiReporter {
+    private val logger =
+        org.gradle.api.logging.Logging.getLogger(InternalA8cCiReporter::class.java)
+
+    fun reportBlocking(
+        metricsReport: MetricsReport,
+        projectName: String,
+        authToken: String?,
     ) {
-        reportLocally(report)
+        runBlocking {
+            report(metricsReport, projectName, authToken)
+        }
+    }
 
-        val payload = report.toAppsMetricsPayload(gradleScanId)
+    suspend fun report(
+        metricsReport: MetricsReport,
+        projectName: String,
+        authToken: String?,
+    ) {
```

Comment:
It is tempting to make some `Report` interface and pass reporter-specific parameters (like `projectName` or `authToken` here) as constructor parameters, as you did in 3bb9601c12c699d47f36036c88bfc60a1faa20e4 with `MetricsReporter`.

I like the current approach with `buildMetricsPreparedAction` more, though, as it's not unnecessarily constraining the implementation. Great job 👍 !

Reply:
Thank you. Implementing a configuration cache can be challenging when using instances, and errors are likely to occur.